java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -142,7 +142,8 @@ ExecutionPath addEdge(ExplodedGraph.Edge edge) {
       flowBuilder.addAll(flow);
       Symbol newTrackSymbol = addFlowFromLearnedAssociations(edge, flowBuilder);
       Stream<Constraint> constraintStream = addFlowFromLearnedConstraints(edge, flowBuilder);
-      return new ExecutionPath(edge, visited.add(edge), newTrackSymbol, flowBuilder.build(), edge.parent.edges().isEmpty() || constraintStream.anyMatch(terminateTraversal));
+      boolean finished = edge.parent.edges().stream().noneMatch(this::notVisited) || constraintStream.anyMatch(terminateTraversal);
```

Comment:
This noneMatch(notVisited) seems rather convoluted for no practical reason : wouldn't a Collections.disjoint be enough in our case ? and/or maybe Guava Sets.intersection  (maybe applicable to sets only) ? Ideally encapsulated in a "visitedAllParents" method ?